packages/core/docs/migrate/2.3.0-rc.1/overview.md

```diff
@@ -0,0 +1,164 @@
+          projectKey: '',
+          clientId: '',
+          clientSecret: '',
+          scopes: []
+        },
+        currency: '',
+        country: ''
+      }
+    }
+  }
+};
+
+```
+
+```javascript
+// middleware.js
+const { createServer } = require('@vue-storefront/middleware');
+const { integrations } = require('./middleware.config');
+
+const app = createServer({ integrations });
+
+app.listen(8181, () => {
+  console.log('Middleware started');
+});
+```
+
+2. Add `'@vue-storefront/middleware/nuxt'` to the `modules` in `nuxt.config.js`.
+
+:::warning Be careful
+Make sure this package is added to the `modules` array, not `buildModules`.
+:::
+
+```javascript
+// nuxt.config.js
+export default {
+  modules: [
+    '@vue-storefront/middleware/nuxt'
+  ]
+};
+```
+
+3. Update custom queries.
+
+Previously, custom queries were callbacks passed as an additional argument in most of our composable methods. Because we moved API communication to the backend, it's no longer possible to pass callbacks from UI (frontend) to `api-client` (backend).
+
+From now on, custom queries are registered in `middleware.config.js` (with the same signature as before), and used on the frontend by providing their unique name:
+
+```javascript{7,17}
+// middleware.config.js
+module.exports = {
+  integrations: {
+    ct: {
+      // other options
+      customQueries: {
+        'MyCustomQuery': (query, variables) => {},
+      }
+    }
+  }
+};
+
+// Some component
+const { search } = useProduct();
+
+onSSR(async () => {
+  await search({ customQuery: { products: 'MyCustomQuery' } })
+});
+```
+
+`customQuery` is now an object that accepts object with:
+- key matching the name of GraphQL query to override,
+- value matching the name of the handler in `middleware.config.js`.
+
+## Checkout
+
+One of the most important pieces missing in VSF Next was Checkout, which originally was supposed to be handled by a single `useCheckout` composable. However, we decided that this approach won't scale well because the checkout process is very different in every integration and it might not work well, if multiple providers will be used to handle different steps of the checkout process, like payment, shipping, etc.
+
+### Changes
+
+We split responsibilities of different steps into multiple new composables:
+
+- `useShipping` for handling shipping information,
+- `useBilling` for handling billing information,
+- `useShippingProvider` for handling shipping providers,
+- `useMakeOrder` for placing the final order.
+
+We also used this opportunity to cleanup other composables:
+- removed `useCheckout`,
+- removed `checkoutGetters`,
+- renamed `useUserOrders` to `useUserOrder` to be consistent with other composables,
+- renamed `isOnCart` and `isOnWishlist` in `useCart` composable to `isInCart` and `isInWishlist`.
+
+These changes affected multiple Vue components. Please refer to [UI section](#ui) for more information.
+
+## Cache
+
+TODO:
+- update Cache library
+
```

Comment:
Waits for https://github.com/vuestorefront/vue-storefront/pull/5610

Reply:
Now it's merged